lib/skate/detours/detours.ex

```diff
@@ -65,34 +67,33 @@ defmodule Skate.Detours.Detours do
           past: list(DetailedDetour.t()) | nil
         }
   def grouped_detours(user_id) do
-    detours =
-      list_detours()
-      |> Enum.map(&db_detour_to_detour/1)
-      |> Enum.filter(& &1)
-      |> Enum.group_by(fn
-        %ActivatedDetourDetails{details: %{status: status}} -> status
-        %{status: status} -> status
-      end)
-
     %{
-      active: Map.get(detours, :active, []),
-      draft:
-        detours
-        |> Map.get(:draft, [])
-        |> Enum.filter(fn detour -> detour.author_id == user_id end),
-      past: Map.get(detours, :past, [])
+      active: detours_for_user(user_id, :active),
+      draft: detours_for_user(user_id, :draft),
+      past: detours_for_user(user_id, :past)
     }
   end
```

Comment:
suggestion: since this is only used in the detours controller, how do you feel about pushing this implementation into the detours controller so we're not having to concern ourselves with keeping this interface available at this level?